sql/timeseries/a11yScores.sql

```diff
@@ -1,24 +1,25 @@
 #standardSQL
-CREATE TEMPORARY FUNCTION getA11yScore(report STRING)
+# Lighthouse changed format of scores in v3.0.0 released in July 2018 so hande old and new
+CREATE TEMPORARY FUNCTION getA11yScore(reportCategories STRING, score STRING)
 RETURNS FLOAT64
 LANGUAGE js AS """
-  $=JSON.parse(report);
-  if ($.reportCategories) {
-    return $.reportCategories.find(i => i.name === 'Accessibility').score;
+  if (reportCategories) {
+    $=JSON.parse(reportCategories);
+    return $.find(i => i.name === 'Accessibility').score;
   } else {
-    return $.categories.accessibility.score * 100;
+    return score * 100;
```

Comment:
Isn't `score` undefined?

Reply:
You either provide the `reportCategories` JSON (for old Lighthouse versions) or the `score` (which is got directly from the JSON_EXTRACT). So no, it shouldn't be undefined here.

Alternative would be to split by date and only call the UDF for the older dates. So have two queries with a `UNION ALL`.

```sql
SELECT
  ...
  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories')...
  ...
WHERE date < XXXX
UNION ALL
SELECT
  ...
  ROUND(APPROX_QUANTILES(JSON_EXTRACT(report, '$.categories.accessibility.score')...
  ...
WHERE date >= XXX
...
```

Reply:
Ah I missed that it was another input param.

One BigQuery'ism that may make this neater is [IFNULL](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#ifnull). Try extracting the score directly and fall back to looking up the category in the UDF when it's null. We could also use an inner SELECT statement to look up the score once per page, then the outer query would do the aggregation for each percentile.

Reply:
Good idea on both counts! Will do that.

Though again gonna play the “it was like this when I got here” card 😉

Reply:
Fixed in https://github.com/HTTPArchive/bigquery/pull/114/commits/8ddc5e6f138587b58c46bc7d991efa58d559b301